test/higher_order_ops/test_invoke_subgraph.py

```diff
@@ -321,6 +321,40 @@ def forward(self, l_x_: "f32[8]", l_y_: "f32[8]"):
 """,
         )
 
+    def test_input_mutation(self):
+        def gn(x, y):
+            x.add_(1)
+            return (torch.mul(x, y),)
+
+        def fn(x, y):
+            return invoke_subgraph(gn, None, (x, y))[0]
+
+        x = torch.randn(8, requires_grad=False)
+        y = torch.randn(8, requires_grad=False)
+
+        opt_fn = torch.compile(fn, backend="inductor", fullgraph=True)
+        with self.assertRaisesRegex(
+            torch._dynamo.exc.Unsupported, "NYI: invoke_subgraph with aliasing"
+        ):
+            opt_fn(x, y)
+
+    def test_input_aliasing(self):
+        def gn(x, y):
+            return (x, torch.mul(x, y))
+
+        def fn(x, y):
+            outs = invoke_subgraph(gn, None, (x, y))
+            return outs[0] * outs[1]
+
+        x = torch.randn(8, requires_grad=False)
+        y = torch.randn(8, requires_grad=False)
+
+        opt_fn = torch.compile(fn, backend="inductor", fullgraph=True)
```

Comment:
nit: these only really need to be backend="eager" (will make the tests run faster), but up to you